pyiron_base/job/template.py

```diff
@@ -35,6 +35,14 @@ def input(self):
     def output(self):
         return self._storage.output
 
+    def to_hdf(self, hdf=None, group_name=None):
+        HasStorage.to_hdf(self, hdf=self.project_hdf5)
+        GenericJob.to_hdf(self, hdf=hdf, group_name=group_name)
```

Comment:
```suggestion
        GenericJob.to_hdf(self, hdf=hdf, group_name=group_name)
        HasStorage.to_hdf(self, hdf=self.project_hdf5)
```
I suspect they should be changed around, because `GenericJob.to_hdf` should be messing with `self.project_hdft`.

Reply:
Yeah, that's how I had it originally then I played around seeing if I could get the job's HDF_VERSION to bump to 0.2.0 by invoking HasStorage first (obviously not, since HasStorage only deals with the storage hdf, it was a stupid idea) -- but it worked both ways in my example and I forgot to switch it back ;) GenericJob first is at the very least more aesthetically pleasing and may have some positive side effect I didn't see.